### PR TITLE
adds json data reading code and removes unnecessary code

### PR DESCRIPTION
**Pull request by visha12l**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,18 +1,44 @@
...
+                </ul>
+                <div className="component" ref='one'>
+                    <CollapseAppIndex />
+                </div>
+                <div className="component hide" ref='two'>
+                    <JsonIndex />
+                </div>
+                <div className="component hide" ref='three'>
+                    <TodoIndex />
+                </div>
```
Comment (avaneeshtripathi) [inline src/App.js]: Its not good to play with hide class. The component remains in the DOM even if it is not needed. We can create a function which will return the desired component. Something like this.

getSubComponent (componentName) {
  switch (componentName) {
    case 'collapseIndex':
      return (
        <div className="component" ref='one'>
          <CollapseAppIndex />
        </div>
      );
    case 'jsonIndex':
      return (
        <div className="component" ref='two'>
          <JsonIndex />
        </div>
    );
    case 'todoIndex':
      return (
        <div className="component" ref='three'>
          <TodoIndex />
        </div>
      );
    default:
      return false;
  }
}

Use a state initially: 
   stateName: ''

And set state on click as:
  this.setState({
    stateName: 'jsonIndex'
  })

And call the function in render() as:
this.getSubComponent(this.state.stateName);

File: src/javascript/readJson/jsonComponent.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import listData from '../../static/colors.json';
+var underscore = require('underscore');
+
+export default class JsonComponent extends React.Component {
+    constructor() {
+        super();
+        this.state={
+            listData: listData.data,
```
Comment (avaneeshtripathi) [inline src/javascript/readJson/jsonComponent.js]: no need to set that in state as we are not updating the data. use listdata directly in map.

Comment (avaneeshtripathi) [reply src/javascript/readJson/jsonComponent.js]: and make this functional component. will shorten the code.

File: src/javascript/containers/jsonIndex.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import Header from '../components/header.js';
+import JsonComponent from '../readJson/jsonComponent.js';
+
+export default class Index extends React.Component {
```
Comment (avaneeshtripathi) [inline src/javascript/containers/jsonIndex.js]: this must be functional component

File: src/App.js (hunk 1 above)
Comment (visha12l) [reply src/App.js]: removed hide class logic

File: src/javascript/containers/jsonIndex.js (hunk 3 above)
Comment (visha12l) [reply src/javascript/containers/jsonIndex.js]: created functional component

File: src/javascript/readJson/jsonComponent.js (hunk 2 above)
Comment (visha12l) [reply src/javascript/readJson/jsonComponent.js]: state removed and created functional component

**Pull request merged**